### ISSUE TITLE
gold All Roots

### ISSUE THREAD

**Issue by darren-yao** [cpinitiative/usaco-guide#869]
module: https://usaco.guide/gold/all-roots?lang=cpp
problem: the given solution is kinda scuffed. i think the better way would be to find the diameter of the tree and run a dfs from the two endpoints of the diameter. here is my (also scuffed) code this with only one dfs function https://cses.fi/paste/2daccc0d7344cd731a4210/

- chessbot108#6397, on discord

**Comment by dongliu0426** [comment]
isn't tree diameter in plat?

**Comment by dolphingarlic** [comment]
Yeah I don't think this is a valid issue because the current solution demonstrates dp on all roots while their solution doesn't

**Comment by bqi343** [comment]
Just add a note that the problem appeared previously and that this is an alternative solution.

**Comment by thecodingwizard** [comment]
Link to previous problem: https://usaco.guide/silver/intro-tree/?lang=cpp#problem-https://cses.fi/problemset/task/1132
